File: BasePlusCommissionEmployee.ts
import { Employees } from './Employees';
interface EmployeeOptions {
  FirstName: string;
  LastName: string;
  SocialSecurityNumber: string;
  grossSales: number;
  commissionRate: number;
  salary: number;
}
class BasePlusCommissionEmployee extends Employees {
  private grossSales!: number;
  private commissionRate!: number;
  private salary!: number;
  constructor(params: EmployeeOptions) {
    super(params);
    this.grossSales = params.grossSales;
    this.commissionRate = params.commissionRate;
    this.salary = params.salary;
  }
  earnings(): number {
    return this.salary + this.grossSales * this.commissionRate;
  }

  raise(percent: number): void {
    this.commissionRate += this.commissionRate * percent;
    this.salary += this.salary * percent;
  }
  toString(): string {
    return `Base Plus Commissioned Employee: ${this.getFirstName()} ${this.getLastName()} with ssn: ${this.getSocialSecurityNumber()}
Gross Sales:${this.grossSales}
Commission Rate:${this.commissionRate}
with Base Salary of: ${this.salary}
Earnings:${this.earnings()}\n`;
  }
}

export { BasePlusCommissionEmployee };

File: src/app/BasePlusCommissionEmployee.java
/*
* To change this license header, choose License Headers in Project Properties.
* To change this template file, choose Tools | Templates
* and open the template in the editor.
*/
package app;

/**
 *
 * @author user
 */
public class BasePlusCommissionEmployee extends Employees {
    private double grossSales;
    private double commissionRate;
    private double salary;

    public BasePlusCommissionEmployee(String FirstName, String LastName, String SocialSecurityNumber, double grossSales,
            double commissionRate, double salary) {
        super(FirstName, LastName, SocialSecurityNumber);
        this.commissionRate = commissionRate;
        this.grossSales = grossSales;
        this.salary = salary;
    }

    @Override
    public String toString() {
        return String.format(
                "Base Salary Plus Commissioned Employee: %s %s with ssn: %s\nGross Sales:%.2f\nCommission Rate:%.4f\nwith Base Salary of: %.2f\nEarnings:$%.2f\n",
                getFirstName(), getLastName(), getSocialSecurityNumber(), grossSales, commissionRate, salary,
                earnings());
    }

    @Override
    public double earnings() {
        return this.salary + this.grossSales * this.commissionRate;
    }

    @Override
    public void raise(double percent) {
        this.commissionRate += this.commissionRate * percent;
        this.salary += this.salary * percent;
    }
}